ntoskrnl/io/iomgr/deviface.c

```diff
@@ -413,25 +413,27 @@ IopOpenInterfaceKey(IN CONST GUID *InterfaceClassGuid,
 }
 
 /*++
- * @name IoGetDeviceInterfaces
+ * @name IopGetDeviceInterfacesWorker
  * @implemented
  *
  * Returns a list of device interfaces of a particular device interface class.
- * Documented in WDK
  *
  * @param InterfaceClassGuid
- *        Points to a class GUID specifying the device interface class
+ *        Points to a class GUID specifying the device interface class.
  *
  * @param PhysicalDeviceObject
  *        Points to an optional PDO that narrows the search to only the
- *        device interfaces of the device represented by the PDO
+ *        device interfaces of the device represented by the PDO.
  *
  * @param Flags
  *        Specifies flags that modify the search for device interfaces. The
  *        DEVICE_INTERFACE_INCLUDE_NONACTIVE flag specifies that the list of
  *        returned symbolic links should contain also disabled device
  *        interfaces in addition to the enabled ones.
  *
+ * @param UseKernelPath
+ *        If TRUE, all the device interfaces in the list will start from \??\.
+ *
  * @param SymbolicLinkList
  *        Points to a character pointer that is filled in on successful return
  *        with a list of unicode strings identifying the device interfaces
```

Comment:
I will try to reformulate this description. The pointer points to a multi-string, basically.

Reply:
"If return status is STATUS_SUCCESS, points to a callee allocated buffer contains null-delimited device interface list. This list ends with two null characters."

Reply:
> This list ends with two null characters

Actually this list ends with an empty string. The reason why there are two null characters, is because the first one belongs to the last (non-empty) NULL-terminated string, and the second NULL is just an empty NULL-terminated string.

Reply:
yep, but this definition can be confusing, Just say that this list has two nulls at the end is much easier.